client/vertex-client.ts

```diff
@@ -69,14 +61,20 @@ export class VertexClient {
   public sceneTemplates: SceneTemplatesApi;
   public translationInspections: TranslationInspectionsApi;
 
-  private basePath: string;
-  private baseOptions: BaseOptions;
   private auth: Oauth2Api;
+  private token: OAuth2Token;
+  private tokenFetchedEpochMs: number;
 
-  private constructor({ auth, baseOptions, basePath, config }: CtorArgs) {
+  private constructor({ auth, baseOptions, basePath, token }: CtorArgs) {
     this.auth = auth;
-    this.baseOptions = baseOptions || {};
-    this.basePath = basePath;
+    this.token = token;
+    this.tokenFetchedEpochMs = nowEpochMs();
+    const config = new Configuration({
+      accessToken: this.accessTokenRefresher,
```

Comment:
The latest generator version allows us to pass a function that returns a promise 😎